radixdlt-java/src/main/java/com/radixdlt/client/core/network/AtomSubmissionUpdate.java

```diff
@@ -57,19 +60,24 @@ public long getTimestamp() {
 		return timestamp;
 	}
 
-	public static AtomSubmissionUpdate now(EUID hid, AtomSubmissionState code) {
-		return new AtomSubmissionUpdate(hid, code, null, System.currentTimeMillis());
+	public AtomSubmissionUpdate putMetaData(String key, Object value) {
+		metaData.put(key, value);
+		return this;
+	}
+
+	public static AtomSubmissionUpdate create(EUID hid, AtomSubmissionState code) {
+		return new AtomSubmissionUpdate(hid, code, null);
 	}
 
-	public static AtomSubmissionUpdate now(EUID hid, AtomSubmissionState code, String message) {
-		return new AtomSubmissionUpdate(hid, code, message, System.currentTimeMillis());
+	public static AtomSubmissionUpdate create(EUID hid, AtomSubmissionState code, String message) {
+		return new AtomSubmissionUpdate(hid, code, message);
 	}
 
 	@Override
 	public String toString() {
 		SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS", Locale.getDefault());
```

Comment:
Could potentially convert to `LocalDateTime` using something like `Instant.ofEpochMilli(millis).atZone(ZoneId.systemDefault()).toLocalDateTime()`, and then using the `toString()` method on the resultant `LocalDateTime` instead of constructing a `SimpleDateFormat` for each call.

Reply:
There reason for using SimpleDateFormat instead of LocalDateTime was due to it being introduced in Java 8 which limits compatibility to Android Oreo API level 26 devices and up.

Currently, the java library goes as far down as Android Nougat API level 24 by avoiding some Java 8 APIs which although not brilliant for maximum compatibility it's a big improvement.

I am aware that there is a backport for the JSR-310 which could be included as a dependency but I wouldn't since it is very inefficient on Android and an adaptation of this was made for use with android specifically as seen here: https://github.com/JakeWharton/ThreeTenABP  

Obviously the Kotlin library aims for 100% compatibility for Android and I would personally prefer at least for now to not make use of LocalDateTime to also be able to maximise the java lib compatibility as much as we can with Android too unless there is a suitable alternative.

Reply:
Ah, OK.  Was not aware of the Android requirement (or even the impact of that).